src/component/feedupdate/index.tsx

```diff
@@ -61,47 +76,94 @@ function FeedUpdate(props: FeedUpdateProps) {
     ...rest
   } = props;
 
-  const { timelineSlug } = useParams();
   const { token } = useContext(AuthContext);
   const ventureId = currentVenture?.id ?? "";
   const timelineId = currentTimeline?.id ?? "";
+  const timelineIds = timelines.map((timeline: ITimeline) => timeline.id);
+  const { mutate: updateUser, status, reset } = useUpdateUser();
 
-  const timelineIds = timelines?.map((timeline: ITimeline) => timeline.id);
-
-  const { data: timelineUpdates = [] } = useUpdatesByTimeline({
-    ventureId,
-    timelineId,
-    token,
-  });
-
-  let updates = timelineUpdates;
+  const { data: timelineUpdates = [], isSuccess: timelineUpdatesSuccess } =
+    useUpdatesByTimeline({
+      ventureId,
+      timelineId,
+      token,
+    });
 
-  const { data: allUpdates = [], isSuccess: updateSuccess } =
+  const { data: ventureUpdates = [], isSuccess: ventureUpdatesSuccess } =
     useUpdatesByTimelineIds({
       ventureId,
       timelineIds,
       token,
     });
 
-  if (updateSuccess) {
-    //deduplicate updates for home
-    const homeUpdates = Array.from(
-      new Set(
-        allUpdates.map((update: IUpdate) =>
-          Math.round(Number(update.id) / 1000000000)
-        )
-      )
-    )
-      .map((id) => {
-        return allUpdates.find(
-          (update: IUpdate) => Math.round(Number(update.id) / 1000000000) === id
-        );
-      })
-      .filter((x): x is IUpdate => Boolean(x));
-
-    // return updates or updates of current timeline.
-    updates = timelineSlug ? timelineUpdates ?? [] : homeUpdates ?? [];
-  }
+  const updates = useMemo<IUpdate[]>(() => {
+    if (timelineId && timelineUpdatesSuccess) {
+      return timelineUpdates;
+    } else if (ventureId && ventureUpdatesSuccess) {
+      return deduplicateUpdates(ventureUpdates);
+    }
+    return [];
+  }, [
+    timelineId,
+    timelineUpdates,
+    timelineUpdatesSuccess,
+    ventureId,
+    ventureUpdates,
+    ventureUpdatesSuccess,
+  ]);
+
+  useEffect(() => {
```

Comment:
Here's the hook that updates the update viewed status for the user. We assume once the update has been rendered here then the user has seen it.